packages/blocks/src/lib/navigation-map/components/nav/obstacles.svelte

```diff
@@ -0,0 +1,152 @@
+<script lang='ts'>
+
+import { IconButton, Label, TextInput } from '@viamrobotics/prime-core';
+import { type LngLat, type Geometry, useMapLibre, type Obstacle, useMapLibreEvent } from '$lib';
+import LnglatInput from '../input/lnglat.svelte';
+import GeometryInputs from '../input/geometry.svelte';
+import OrientationInput from '../input/orientation.svelte';
+import { environment, hovered, boundingRadius, obstacles } from '../../stores';
+import { calculateBoundingBox } from '../../lib/bounding-box';
+import { createEventDispatcher } from 'svelte';
+import { createObstacle } from '$lib/navigation-map/lib/create-obstacle';
+import { createName } from '$lib/navigation-map/lib/create-name';
+
+type Events = {
+  update: Obstacle[]
+};
+
+const dispatch = createEventDispatcher<Events>();
+
+const { map } = useMapLibre();
+
+const handleSelect = (selection: { name: string; location: LngLat }) => {
+  const zoom = boundingRadius[selection.name]!;
+  const bb = calculateBoundingBox(zoom, selection.location);
+  map.fitBounds(bb, { duration: 800, curve: 0.1 });
+};
+
+const handleLngLatInput = (name: string) => (event: CustomEvent<LngLat>) => {
+  const index = $obstacles.findIndex((obstacle) => obstacle.name === name);
```

Comment:
Here and in the other handlers: is it safe to assume that the obstacle index will be found? You assert that the obstacles exist immediately after finding index.

Reply:
Yep, name is a required key for this data structure, so an error being thrown if an index isn't found makes sense to me